lib/discourse.rb

```diff
@@ -113,16 +113,19 @@ def self.activate_plugins!
   end
 
   def self.recently_readonly?
-    return false unless @last_read_only
-    @last_read_only > 15.seconds.ago
+    recently_readonly = $redis.get(recently_readonly_mode_key)
```

Comment:
This implementation though now means that if redis gets a readonly error it will not unconditionally do a `get` every time. so every write becomes, write + read 

I think probably better to use a hash here and key on multisite db name.